continue string-vs-str (#1133)
exercism/rust: exercises/concept/string-vs-str/.docs/instructions.md

> Your company has been hired to generate CSV documents from user input.

Reviewer: This looks like an unintentional semantic change: the builder produces individual CSV records, not full documents.

```suggestion
Your company has been hired to generate CSV records from user input.
```

While a formal specification exists, your company only needs to implement the following functionality:

> - create a new builder
> - add an individual record
> - serialize records with basic escaping

Reviewer: ```suggestion
- create a new record builder
- add an individual field to the record
- serialize the record with basic escaping
```

Precision in wording.

## 1. Create a new builder

You'll need to implement the `new` method to create a new builder.

```rust
let mut builder = CSVRecordBuilder::new();
```

> ## 2. Add individual records

Reviewer: ```suggestion
## 2. Add individual fields to the record.
```

> You'll need to implement the `add` method to append records to your builder.

Reviewer: ```suggestion
You'll need to implement the `add` method to append fields to your record builder.
```

You'll need to implement the `add` method to append records to your builder.
`add` takes immutable string slices, `&str`, as the argument.

```rust
builder.add("ant");
builder.add("ba\"t");
builder.add("cat");

let list = builder.build();
//Note that from now on we can do nothing now with builder.
assert_eq!(r#"ant,"ba""t",cat"#, &list);
```

> ## Edge cases

Reviewer: It would be nice to retain this section's information somewhere, so students don't actually have to chase down the necessary rules in the formal specification. Maybe `about.md`?

Author: Great idea. I will preserve it there!

> ## 3. Serialize records with basic escaping

Reviewer: ```suggestion
## 3. Serialize the record with basic escaping
```

> let list = builder.build();
> // Note that from now on we cannot use `builder`
> assert_eq!(r#"ant,"ba""t",cat"#, &list);

Reviewer: ```suggestion
let record = builder.build();
// Note that from now on we cannot use `builder`
assert_eq!(&record, r#"ant,"ba""t",cat"#);
```

A more useful name.
